usecase/pokemon.go

```diff
@@ -129,3 +131,81 @@ func (u PokemonUsecase) DeletePokemon(id string) (model.Pokemon, error) {
 
 	return pokemon, err
 }
+
+func (u PokemonUsecase) GetConcurrentPokemonCSV(types string, items string, itemsPerWorker string) (model.PokemonList, error) {
+	var pokemonList model.PokemonList
+	var err error
+
+	itemsInt, _ := strconv.Atoi(items)
+	itemsPerWorkerInt, err := strconv.Atoi(itemsPerWorker)
+	poolSize := utils.GetIntFromDivision(itemsInt, itemsPerWorkerInt)
+	limit := utils.GetIntFromDivision(len(pokemonList), poolSize)
+
+	fmt.Println(poolSize)
+	fmt.Println(limit)
+
+	if err == nil {
+		pokemonList, err := u.c.ReadCSV()
+		var oddPokemonList []model.Pokemon
+		var evenPokemonList []model.Pokemon
+
+		if err != nil {
+			return pokemonList, err
+		}
+
+		chOd := make(chan int)
+		chEv := make(chan int)
+
+		var wg sync.WaitGroup
+		wg.Add(poolSize)
+
+		for i := 0; i < poolSize; i++ {
+			go func(i int) {
+				tmpPokemon := pokemonList[i]
+				if tmpPokemon.Id%2 != 0 {
+					chOd <- tmpPokemon.Id
+				} else if tmpPokemon.Id%2 != 1 {
```

Comment:
I don't think you need to add this second if.
If the number is not odd then it is even, no need to double check it

Reply:
My mistake, I forgot to delete this unnecessary validation, thanks!